browser/gemini/gemini_service_factory.cc

```diff
@@ -0,0 +1,41 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/gemini/gemini_service_factory.h"
+
+#include "brave/components/gemini/browser/gemini_service.h"
+#include "chrome/browser/profiles/incognito_helpers.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/keyed_service/content/browser_context_dependency_manager.h"
+
+// static
+GeminiServiceFactory* GeminiServiceFactory::GetInstance() {
+  return base::Singleton<GeminiServiceFactory>::get();
+}
+
+// static
+GeminiService* GeminiServiceFactory::GetForProfile(Profile* profile) {
```

Comment:
it seems that in every case you are casting to the Profile so it's better to have GetForBrowserContext